sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1543,6 +1544,90 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResults<Application>> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "currentAssessment.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "application.applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest",
+      )
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where(
+        new Brackets((qb) =>
+          qb
+            .where("applicationOfferingChangeRequest.id IS NULL")
+            .orWhere(
+              "applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus NOT IN (:...status)",
+              {
+                status: [
+                  ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+                  ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+                ],
+              },
+            ),
+        ),
+      )
+      .andWhere("application.location.id = :locationId", { locationId })
+      .andWhere("application.applicationStatus = :applicationStatus", {
+        applicationStatus: ApplicationStatus.Completed,
+      })
+      .andWhere("application.isArchived = false");
```

Comment:
For my understanding, after the disbursements for the application are processed, does the application go to the archived state?

Reply:
there is an archived day configured, so, once a completed application is past that day from the offering's last day, the application will be archived, and another way application gets archived is when there is a scholastic standing happening for an application.
Reference
https://vscode.dev/github/bcgov/SIMS/blob/feature/sims-1953/part-2/sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts#L26
https://vscode.dev/github/bcgov/SIMS/blob/feature/sims-1953/part-2/sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts#L251

Reply:
Thanks @ann-aot